prow/plugins/plugins.go

```diff
@@ -100,15 +100,72 @@ type PluginClient struct {
 	GitClient    *git.Client
 	SlackClient  *slack.Client
 	Config       *config.Config
```

Comment:
This is getting a bit confusing. Can you figure out how exactly this is used by plugins and add a comment to that extend?

Reply:
Hmm, plugins may not need it, so I will try to remove it.

Reply:
Looks like `trigger` needs presubmits and postsubmits. Makes sense to continue to expose job config to plugins but maybe not stuff for e.g. tide. I could create

```go
type JobConfiguration struct {
	Presubmits  map[string][]Presubmit
	Postsubmits map[string][]Postsubmit
	Periodics []Periodic
}
```

And make that available to plugins instead of the entire config?

Reply:
No need to do this here. At least add a comment now about its usage.

Reply:
@kargakis done